hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Desugarer.scala

```diff
@@ -478,19 +479,27 @@ class Desugarer(val elaborator: Elaborator)
       val clsTrm = elaborator.cls(ctor, inAppPrefix = false)
       clsTrm.symbol.flatMap(_.asClsLike) match
       case S(cls: ClassSymbol) =>
-        val arity = cls.arity
-        if arity =/= args.length then
-          val m = args.length.toString
-          error:
-            if arity == 0 then
-              msg"the constructor does not take any arguments but found $m" -> app.toLoc
-            else
-              msg"mismatched arity: expect ${arity.toString}, found $m" -> app.toLoc
-        val params = scrutSymbol.getSubScrutinees(cls)
+        val paramSymbols = cls.defn match
+          case S(Parameterized(_, _, _, _, _, paramList, _, _, _, _)) =>
```

Comment:
@chengluyu FYI this is an anti-pattern. Who wants to freaking count the number of underscores?